File: sequential-or-linear-searching/linear-search/LinearSearch.py
def linear_search(arr, target):
    for index, value in enumerate(arr):
        if value == target:
            return index
    return -1


def main():
    arr = [12, 7, 15, 9, 21]
    target = 9

    index = linear_search(arr, target)

    if index != -1:
        print(f"Element found at index: {index}")
    else:
        print("Element not found")


if __name__ == "__main__":
    main()

File: sequential-or-linear-searching/linear-search/LinearSearch.js
function linearSearch(arr, target) {
    for (let i = 0; i < arr.length; i++) {
        if (arr[i] === target) {
            return i;
        }
    }
    return -1;
}

function main() {
    const arr = [12, 7, 15, 9, 21];
    const target = 9;

    const index = linearSearch(arr, target);

    if (index !== -1) {
        console.log(`Element found at index: ${index}`);
    } else {
        console.log("Element not found");
    }
}

main();
